lib/rspec/core/metadata.rb

```diff
@@ -173,6 +181,12 @@ def build_description_from(parent_description=nil, my_description=nil)
           (parent_description.to_s + separator) << my_description.to_s
         end
 
+        def build_scoped_id_for(file_path)
+          index = @index_provider.call(file_path).to_s
+          parent_scoped_id = metadata.fetch(:scoped_id) { return index }
```

Comment:
It's very much necessary.  It's an early return of `index` from `build_scoped_id_for`.  Remember that `return` is relative to the enclosing method definition since this is a block, not a lambda.